### PR TITLE
Add access to the environment variables from configs

### PR DESCRIPTION
**Pull request by imnetcat**
#1466 solution option
now we can write in the config like 
```
...
this.PORT || 8000
...
```

### CODE REVIEW

File: lib/worker.js [hunk 1]
```diff
@@ -10,8 +10,12 @@ const { Server } = metarhia.metacom;
 const { Auth } = require('./auth.js');
 
 (async () => {
+  const vm = require('vm');
```
Comment (tshemsedinov) [inline lib/worker.js:13]:
```suggestion
```
All modules are loaded, see `dependencies.js`

File: lib/worker.js [hunk 2]
```diff
@@ -10,8 +10,12 @@ const { Server } = metarhia.metacom;
...
+  const sandbox = { PORT: parseInt(process.env.PORT || 0) };
```
Comment (tshemsedinov) [inline lib/worker.js:14]:
```suggestion
  const sandbox = { process };
```
Not a good place for parsing ports, we can just pass process into sandbox here.

File: lib/worker.js [hunk 3]
```diff
@@ -10,8 +10,12 @@ const { Server } = metarhia.metacom;
...
+  vm.createContext(sandbox);
+
```
Comment (tshemsedinov) [inline lib/worker.js:16]:
```suggestion
  const context = metavm.createContext(sandbox);
```
Use metavm here

File: lib/worker.js [hunk 4]
```diff
@@ -10,8 +10,12 @@ const { Server } = metarhia.metacom;
...
   const configPath = path.join(application.path, 'config');
-  const options = { mode: process.env.MODE };
+  const options = { mode: process.env.MODE, sandbox };
```
Comment (tshemsedinov) [inline lib/worker.js:18]:
```suggestion
  const options = { mode: process.env.MODE, context };
```
We will rename sandbox to context in `metarhia/config` first, so `sandbox` will be used just in `impress` or other upper-level code but both `metavm` and `metarhia/config` will use `{ context }` in `options`

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
We will change it in a different way

**Comment by imnetcat** [comment]
Okey, great

**Comment by tshemsedinov** [comment]
Will be landed after this fix https://github.com/metarhia/config/issues/67

**Comment by tshemsedinov** [comment]
To be landed after https://github.com/metarhia/impress/pull/1468

File: lib/worker.js [hunk 5]
```diff
@@ -10,8 +10,10 @@ const { Server } = metarhia.metacom;
 const { Auth } = require('./auth.js');
 
 (async () => {
+  const sandbox = { process };
+  const context = metarhia.metavm.createContext(sandbox);
```
Comment (tshemsedinov) [inline lib/worker.js:14]:
```suggestion
  const context = metarhia.metavm.createContext({ process });
```

File: lib/worker.js [hunk 6]
```diff
@@ -10,8 +10,9 @@ const { Server } = metarhia.metacom;
 const { Auth } = require('./auth.js');
 
 (async () => {
+  const context = metarhia.metavm.createContext({ process });
   const configPath = path.join(application.path, 'config');
```
Comment (tshemsedinov) [inline lib/worker.js:14]:
```suggestion
  const configPath = path.join(application.path, 'config');
  const context = metarhia.metavm.createContext({ process });
```

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/impress/commit/4c4098802de80be220bf955eab64e9492679456c

**Pull request closed without merging**